compiler/cmd/src/compile_bpf.rs

```diff
@@ -30,7 +30,7 @@ fn compile_bpf_object(args: &Options, source_path: &str, output_path: &str) -> R
     let target_arch = get_target_arch(&args.compile_opts)?;
 
     let command = format!(
-        "{} -g -O2 -target bpf -Wno-unknown-attributes -D__TARGET_ARCH_{} {} {} {} {} -c {} -o {}",
+        "{} -g -O2 -fstack-protector -target bpf -Wno-unknown-attributes -D__TARGET_ARCH_{} {} {} {} {} -c {} -o {}",
```

Comment:
This flag has caused some issues, but this flag will affect building with clang on specific Linux distros.

Reply:
Maybe we should have a flags lists instead of hard code it here?

For example, a vec contains all build flags needed, include -O2 -g -target bpf, and some distribution specify flags like this. The vec will be flattened into a string here.

Reply:
Thanks, I will complete this change later.